### PR TITLE
build: Update setup to remove usage of threefive

### PR DESCRIPTION
**Pull request by airlangga-m2amedia**
When I included threefive in one of my projects, I came across this error:
```
11:02:42  #22 3.252   �� python setup.py egg_info did not run successfully.
11:02:42  #22 3.252   ��� exit code: 1
11:02:42  #22 3.252   ������> [12 lines of output]
11:02:42  #22 3.252       Traceback (most recent call last):
11:02:42  #22 3.252         File "<string>", line 2, in <module>
11:02:42  #22 3.252         File "<pip-setuptools-caller>", line 34, in <module>
11:02:42  #22 3.252         File "/tmp/pip-req-build-r1x2ys8b/setup.py", line 4, in <module>
11:02:42  #22 3.252           import threefive
11:02:42  #22 3.252         File "/tmp/pip-req-build-r1x2ys8b/threefive/__init__.py", line 6, in <module>
11:02:42  #22 3.252           from .decode import decode
11:02:42  #22 3.252         File "/tmp/pip-req-build-r1x2ys8b/threefive/decode.py", line 24, in <module>
11:02:42  #22 3.252           from .stream import Stream
11:02:42  #22 3.252         File "/tmp/pip-req-build-r1x2ys8b/threefive/stream.py", line 6, in <module>
11:02:42  #22 3.252           from new_reader import reader
11:02:42  #22 3.252       ModuleNotFoundError: No module named 'new_reader'
11:02:42  #22 3.252       [end of output]
11:02:42  #22 3.252   
11:02:42  #22 3.252   note: This error originates from a subprocess, and is likely not a problem with pip.
11:02:42  #22 3.254 error: metadata-generation-failed
```

I traced this error back to the use of `import threefive` in the setup.py file. I noticed it was being used to set the current version of the module.
This PR removes the use of the threefive module in setup.py, instead replacing it with a static file called VERSION in which the version is updated. The code in the version module has been updated to use this static file.

### CODE REVIEW

**Comment by airlangga-m2amedia** [comment]
It now also explicitly sets the encoding on any invocation of `open()` for the benefit of our Windows brethren.

**Comment by airlangga-m2amedia** [comment]
Proof that the `make install` command works, where it did not before:
```
make install
rm -f dist/*
rm -rf build/*
python3 setup.py sdist bdist_wheel
running sdist
running egg_info
creating threefive.egg-info
writing threefive.egg-info/PKG-INFO
writing dependency_links to threefive.egg-info/dependency_links.txt
writing requirements to threefive.egg-info/requires.txt
writing top-level names to threefive.egg-info/top_level.txt
writing manifest file 'threefive.egg-info/SOURCES.txt'
reading manifest file 'threefive.egg-info/SOURCES.txt'
adding license file 'LICENSE'
writing manifest file 'threefive.egg-info/SOURCES.txt'
running check
creating threefive-2.3.75
creating threefive-2.3.75/bin
creating threefive-2.3.75/threefive
creating threefive-2.3.75/threefive.egg-info
copying files to threefive-2.3.75...
copying LICENSE -> threefive-2.3.75
copying README.md -> threefive-2.3.75
copying setup.py -> threefive-2.3.75
copying bin/threefive -> threefive-2.3.75/bin
copying threefive/__init__.py -> threefive-2.3.75/threefive
copying threefive/base.py -> threefive-2.3.75/threefive
copying threefive/bitn.py -> threefive-2.3.75/threefive
copying threefive/commands.py -> threefive-2.3.75/threefive
copying threefive/crc.py -> threefive-2.3.75/threefive
copying threefive/cue.py -> threefive-2.3.75/threefive
copying threefive/decode.py -> threefive-2.3.75/threefive
copying threefive/descriptors.py -> threefive-2.3.75/threefive
copying threefive/encode.py -> threefive-2.3.75/threefive
copying threefive/packetdata.py -> threefive-2.3.75/threefive
copying threefive/section.py -> threefive-2.3.75/threefive
copying threefive/segment.py -> threefive-2.3.75/threefive
copying threefive/segmentation.py -> threefive-2.3.75/threefive
copying threefive/smoketest.py -> threefive-2.3.75/threefive
copying threefive/stream.py -> threefive-2.3.75/threefive
copying threefive/stuff.py -> threefive-2.3.75/threefive
copying threefive/upids.py -> threefive-2.3.75/threefive
copying threefive/version.py -> threefive-2.3.75/threefive
copying threefive.egg-info/PKG-INFO -> threefive-2.3.75/threefive.egg-info
copying threefive.egg-info/SOURCES.txt -> threefive-2.3.75/threefive.egg-info
copying threefive.egg-info/dependency_links.txt -> threefive-2.3.75/threefive.egg-info
copying threefive.egg-info/requires.txt -> threefive-2.3.75/threefive.egg-info
copying threefive.egg-info/top_level.txt -> threefive-2.3.75/threefive.egg-info
Writing threefive-2.3.75/setup.cfg
creating dist
Creating tar archive
removing 'threefive-2.3.75' (and everything under it)
running bdist_wheel
running build
running build_py
creating build
creating build/lib
creating build/lib/threefive
copying threefive/stuff.py -> build/lib/threefive
copying threefive/crc.py -> build/lib/threefive
copying threefive/encode.py -> build/lib/threefive
copying threefive/version.py -> build/lib/threefive
copying threefive/bitn.py -> build/lib/threefive
copying threefive/smoketest.py -> build/lib/threefive
copying threefive/__init__.py -> build/lib/threefive
copying threefive/upids.py -> build/lib/threefive
copying threefive/section.py -> build/lib/threefive
copying threefive/stream.py -> build/lib/threefive
copying threefive/descriptors.py -> build/lib/threefive
copying threefive/packetdata.py -> build/lib/threefive
copying threefive/segment.py -> build/lib/threefive
copying threefive/decode.py -> build/lib/threefive
copying threefive/commands.py -> build/lib/threefive
copying threefive/base.py -> build/lib/threefive
copying threefive/cue.py -> build/lib/threefive
copying threefive/segmentation.py -> build/lib/threefive
running build_scripts
creating build/scripts-3.9
copying and adjusting bin/threefive -> build/scripts-3.9
changing mode of build/scripts-3.9/threefive from 644 to 755
/Users/airlangga/Documents/git/scte35-threefive/venv/lib/python3.9/site-packages/setuptools/command/install.py:34: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.
  warnings.warn(
installing to build/bdist.macosx-13.2-arm64/wheel
running install
running install_lib
creating build/bdist.macosx-13.2-arm64
creating build/bdist.macosx-13.2-arm64/wheel
creating build/bdist.macosx-13.2-arm64/wheel/threefive
copying build/lib/threefive/stuff.py -> build/bdist.macosx-13.2-arm64/wheel/threefive
copying build/lib/threefive/crc.py -> build/bdist.macosx-13.2-arm64/wheel/threefive
copying build/lib/threefive/encode.py -> build/bdist.macosx-13.2-arm64/wheel/threefive
copying build/lib/threefive/version.py -> build/bdist.macosx-13.2-arm64/wheel/threefive
copying build/lib/threefive/bitn.py -> build/bdist.macosx-13.2-arm64/wheel/threefive
copying build/lib/threefive/smoketest.py -> build/bdist.macosx-13.2-arm64/wheel/threefive
copying build/lib/threefive/__init__.py -> build/bdist.macosx-13.2-arm64/wheel/threefive
copying build/lib/threefive/upids.py -> build/bdist.macosx-13.2-arm64/wheel/threefive
copying build/lib/threefive/section.py -> build/bdist.macosx-13.2-arm64/wheel/threefive
copying build/lib/threefive/stream.py -> build/bdist.macosx-13.2-arm64/wheel/threefive
copying build/lib/threefive/descriptors.py -> build/bdist.macosx-13.2-arm64/wheel/threefive
copying build/lib/threefive/packetdata.py -> build/bdist.macosx-13.2-arm64/wheel/threefive
copying build/lib/threefive/segment.py -> build/bdist.macosx-13.2-arm64/wheel/threefive
copying build/lib/threefive/decode.py -> build/bdist.macosx-13.2-arm64/wheel/threefive
copying build/lib/threefive/commands.py -> build/bdist.macosx-13.2-arm64/wheel/threefive
copying build/lib/threefive/base.py -> build/bdist.macosx-13.2-arm64/wheel/threefive
copying build/lib/threefive/cue.py -> build/bdist.macosx-13.2-arm64/wheel/threefive
copying build/lib/threefive/segmentation.py -> build/bdist.macosx-13.2-arm64/wheel/threefive
running install_egg_info
Copying threefive.egg-info to build/bdist.macosx-13.2-arm64/wheel/threefive-2.3.75-py3.9.egg-info
running install_scripts
creating build/bdist.macosx-13.2-arm64/wheel/threefive-2.3.75.data
creating build/bdist.macosx-13.2-arm64/wheel/threefive-2.3.75.data/scripts
copying build/scripts-3.9/threefive -> build/bdist.macosx-13.2-arm64/wheel/threefive-2.3.75.data/scripts
changing mode of build/bdist.macosx-13.2-arm64/wheel/threefive-2.3.75.data/scripts/threefive to 755
creating build/bdist.macosx-13.2-arm64/wheel/threefive-2.3.75.dist-info/WHEEL
creating 'dist/threefive-2.3.75-py3-none-any.whl' and adding 'build/bdist.macosx-13.2-arm64/wheel' to it
adding 'threefive/__init__.py'
adding 'threefive/base.py'
adding 'threefive/bitn.py'
adding 'threefive/commands.py'
adding 'threefive/crc.py'
adding 'threefive/cue.py'
adding 'threefive/decode.py'
adding 'threefive/descriptors.py'
adding 'threefive/encode.py'
adding 'threefive/packetdata.py'
adding 'threefive/section.py'
adding 'threefive/segment.py'
adding 'threefive/segmentation.py'
adding 'threefive/smoketest.py'
adding 'threefive/stream.py'
adding 'threefive/stuff.py'
adding 'threefive/upids.py'
adding 'threefive/version.py'
adding 'threefive-2.3.75.data/scripts/threefive'
adding 'threefive-2.3.75.dist-info/LICENSE'
adding 'threefive-2.3.75.dist-info/METADATA'
adding 'threefive-2.3.75.dist-info/WHEEL'
adding 'threefive-2.3.75.dist-info/top_level.txt'
adding 'threefive-2.3.75.dist-info/RECORD'
removing build/bdist.macosx-13.2-arm64/wheel
python3  setup.py install
running install
/Users/airlangga/Documents/git/scte35-threefive/venv/lib/python3.9/site-packages/setuptools/command/install.py:34: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.
  warnings.warn(
/Users/airlangga/Documents/git/scte35-threefive/venv/lib/python3.9/site-packages/setuptools/command/easy_install.py:144: EasyInstallDeprecationWarning: easy_install command is deprecated. Use build and pip and other standards-based tools.
  warnings.warn(
running bdist_egg
running egg_info
writing threefive.egg-info/PKG-INFO
writing dependency_links to threefive.egg-info/dependency_links.txt
writing requirements to threefive.egg-info/requires.txt
writing top-level names to threefive.egg-info/top_level.txt
reading manifest file 'threefive.egg-info/SOURCES.txt'
adding license file 'LICENSE'
writing manifest file 'threefive.egg-info/SOURCES.txt'
installing library code to build/bdist.macosx-13.2-arm64/egg
running install_lib
running build_py
creating build/bdist.macosx-13.2-arm64/egg
creating build/bdist.macosx-13.2-arm64/egg/threefive
copying build/lib/threefive/stuff.py -> build/bdist.macosx-13.2-arm64/egg/threefive
copying build/lib/threefive/crc.py -> build/bdist.macosx-13.2-arm64/egg/threefive
copying build/lib/threefive/encode.py -> build/bdist.macosx-13.2-arm64/egg/threefive
copying build/lib/threefive/version.py -> build/bdist.macosx-13.2-arm64/egg/threefive
copying build/lib/threefive/bitn.py -> build/bdist.macosx-13.2-arm64/egg/threefive
copying build/lib/threefive/smoketest.py -> build/bdist.macosx-13.2-arm64/egg/threefive
copying build/lib/threefive/__init__.py -> build/bdist.macosx-13.2-arm64/egg/threefive
copying build/lib/threefive/upids.py -> build/bdist.macosx-13.2-arm64/egg/threefive
copying build/lib/threefive/section.py -> build/bdist.macosx-13.2-arm64/egg/threefive
copying build/lib/threefive/stream.py -> build/bdist.macosx-13.2-arm64/egg/threefive
copying build/lib/threefive/descriptors.py -> build/bdist.macosx-13.2-arm64/egg/threefive
copying build/lib/threefive/packetdata.py -> build/bdist.macosx-13.2-arm64/egg/threefive
copying build/lib/threefive/segment.py -> build/bdist.macosx-13.2-arm64/egg/threefive
copying build/lib/threefive/decode.py -> build/bdist.macosx-13.2-arm64/egg/threefive
copying build/lib/threefive/commands.py -> build/bdist.macosx-13.2-arm64/egg/threefive
copying build/lib/threefive/base.py -> build/bdist.macosx-13.2-arm64/egg/threefive
copying build/lib/threefive/cue.py -> build/bdist.macosx-13.2-arm64/egg/threefive
copying build/lib/threefive/segmentation.py -> build/bdist.macosx-13.2-arm64/egg/threefive
byte-compiling build/bdist.macosx-13.2-arm64/egg/threefive/stuff.py to stuff.cpython-39.pyc
byte-compiling build/bdist.macosx-13.2-arm64/egg/threefive/crc.py to crc.cpython-39.pyc
byte-compiling build/bdist.macosx-13.2-arm64/egg/threefive/encode.py to encode.cpython-39.pyc
byte-compiling build/bdist.macosx-13.2-arm64/egg/threefive/version.py to version.cpython-39.pyc
byte-compiling build/bdist.macosx-13.2-arm64/egg/threefive/bitn.py to bitn.cpython-39.pyc
byte-compiling build/bdist.macosx-13.2-arm64/egg/threefive/smoketest.py to smoketest.cpython-39.pyc
byte-compiling build/bdist.macosx-13.2-arm64/egg/threefive/__init__.py to __init__.cpython-39.pyc
byte-compiling build/bdist.macosx-13.2-arm64/egg/threefive/upids.py to upids.cpython-39.pyc
byte-compiling build/bdist.macosx-13.2-arm64/egg/threefive/section.py to section.cpython-39.pyc
byte-compiling build/bdist.macosx-13.2-arm64/egg/threefive/stream.py to stream.cpython-39.pyc
byte-compiling build/bdist.macosx-13.2-arm64/egg/threefive/descriptors.py to descriptors.cpython-39.pyc
byte-compiling build/bdist.macosx-13.2-arm64/egg/threefive/packetdata.py to packetdata.cpython-39.pyc
byte-compiling build/bdist.macosx-13.2-arm64/egg/threefive/segment.py to segment.cpython-39.pyc
byte-compiling build/bdist.macosx-13.2-arm64/egg/threefive/decode.py to decode.cpython-39.pyc
byte-compiling build/bdist.macosx-13.2-arm64/egg/threefive/commands.py to commands.cpython-39.pyc
byte-compiling build/bdist.macosx-13.2-arm64/egg/threefive/base.py to base.cpython-39.pyc
byte-compiling build/bdist.macosx-13.2-arm64/egg/threefive/cue.py to cue.cpython-39.pyc
byte-compiling build/bdist.macosx-13.2-arm64/egg/threefive/segmentation.py to segmentation.cpython-39.pyc
creating build/bdist.macosx-13.2-arm64/egg/EGG-INFO
installing scripts to build/bdist.macosx-13.2-arm64/egg/EGG-INFO/scripts
running install_scripts
running build_scripts
creating build/bdist.macosx-13.2-arm64/egg/EGG-INFO/scripts
copying build/scripts-3.9/threefive -> build/bdist.macosx-13.2-arm64/egg/EGG-INFO/scripts
changing mode of build/bdist.macosx-13.2-arm64/egg/EGG-INFO/scripts/threefive to 755
copying threefive.egg-info/PKG-INFO -> build/bdist.macosx-13.2-arm64/egg/EGG-INFO
copying threefive.egg-info/SOURCES.txt -> build/bdist.macosx-13.2-arm64/egg/EGG-INFO
copying threefive.egg-info/dependency_links.txt -> build/bdist.macosx-13.2-arm64/egg/EGG-INFO
copying threefive.egg-info/requires.txt -> build/bdist.macosx-13.2-arm64/egg/EGG-INFO
copying threefive.egg-info/top_level.txt -> build/bdist.macosx-13.2-arm64/egg/EGG-INFO
zip_safe flag not set; analyzing archive contents...
creating 'dist/threefive-2.3.75-py3.9.egg' and adding 'build/bdist.macosx-13.2-arm64/egg' to it
removing 'build/bdist.macosx-13.2-arm64/egg' (and everything under it)
Processing threefive-2.3.75-py3.9.egg
Copying threefive-2.3.75-py3.9.egg to /Users/airlangga/Documents/git/scte35-threefive/venv/lib/python3.9/site-packages
Adding threefive 2.3.75 to easy-install.pth file
Installing threefive script to /Users/airlangga/Documents/git/scte35-threefive/venv/bin

Installed /Users/airlangga/Documents/git/scte35-threefive/venv/lib/python3.9/site-packages/threefive-2.3.75-py3.9.egg
Processing dependencies for threefive==2.3.75
Searching for pyaes
Reading https://pypi.org/simple/pyaes/
Downloading https://files.pythonhosted.org/packages/44/66/2c17bae31c906613795711fc78045c285048168919ace2220daa372c7d72/pyaes-1.6.1.tar.gz#sha256=02c1b1405c38d3c370b085fb952dd8bea3fadcee6411ad99f312cc129c536d8f
Best match: pyaes 1.6.1
Processing pyaes-1.6.1.tar.gz
Writing /var/folders/v5/qzrj3x0n6d33gxy3x07zclrc0000gn/T/easy_install-zepj8tpl/pyaes-1.6.1/setup.cfg
Running pyaes-1.6.1/setup.py -q bdist_egg --dist-dir /var/folders/v5/qzrj3x0n6d33gxy3x07zclrc0000gn/T/easy_install-zepj8tpl/pyaes-1.6.1/egg-dist-tmp-8javgcrn
/Users/airlangga/Documents/git/scte35-threefive/venv/lib/python3.9/site-packages/setuptools/command/install.py:34: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.
  warnings.warn(
zip_safe flag not set; analyzing archive contents...
Adding pyaes 1.6.1 to easy-install.pth file

Installed /Users/airlangga/Documents/git/scte35-threefive/venv/lib/python3.9/site-packages/pyaes-1.6.1-py3.9.egg
Searching for new_reader>=0.1.3
Reading https://pypi.org/simple/new_reader/
Downloading https://files.pythonhosted.org/packages/9f/ed/652308c0f35bf5564c5da84736d0bf5a16e905366513577221157c601c69/new_reader-0.1.7-py3-none-any.whl#sha256=24f90eab377ff72a7f41f1a9ef4dbeba7b1669aa9dfe1c71a641545d97dac0b7
Best match: new-reader 0.1.7
Processing new_reader-0.1.7-py3-none-any.whl
Installing new_reader-0.1.7-py3-none-any.whl to /Users/airlangga/Documents/git/scte35-threefive/venv/lib/python3.9/site-packages
Adding new-reader 0.1.7 to easy-install.pth file

Installed /Users/airlangga/Documents/git/scte35-threefive/venv/lib/python3.9/site-packages/new_reader-0.1.7-py3.9.egg
Finished processing dependencies for threefive==2.3.75
```

File: threefive/version.py [hunk 1]
```diff
@@ -1,30 +1,34 @@
...
+any current consumer of the version module.
 """
 
-MAJOR = "2"
-MINOR = "3"
-MAINTAINENCE = "75"
-
 
-def version():
+def version() -> str:
```
Comment (futzu) [inline threefive/version.py:19]: I have been expecting somebody to push me toward the newer style functions, I need to start using them.

File: threefive/version.py [hunk 2]
```diff
@@ -1,30 +1,34 @@
...
 
-def version_number():
+def version_number() -> int:
     """
     version_number returns version as an int.
     if version() returns 2.3.01
     version_number will return 2301
     """
-    return int(f"{MAJOR}{MINOR}{MAINTAINENCE}")
+    return int(version().replace(".", ""))
```
Comment (futzu) [inline threefive/version.py:34]: DRY, makes sense.

**Review by futzu — APPROVED** [review]
Where does the VERSION file go on a pip install?
I guess we'll find out :)

**Comment by futzu** [comment]
```py3
>>>> from threefive import version
>>>> version()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/a/.local/lib/pypy3.9/site-packages/threefive/version.py", line 23, in version
    with open("VERSION", "r", encoding="utf-8") as ver:
FileNotFoundError: [Errno 2] No such file or directory: 'VERSION'
```

**Comment by futzu** [comment]
What if we just did 
```
from  threefive.version import version 
```
in the setup file to get around the new_reader dependency?

Either that , or we move VERSION into the threefive directory or something.

**Comment by futzu** [comment]
I think my whole concept of version.py is wrong. I am thinking we remove version.py entirely for now, and just go with your VERSION file. I am going to do that right now.

**Comment by futzu** [comment]
I got an idea.

```py3
#!/usr/bin/env python3

import setuptools

with open("README.md", "r", encoding="utf-8") as fh:
    readme = fh.read()

with open("VERSION", "r", encoding="utf-8") as ver:
    version = ver.read()

# How about this?
with open("threefive/latest.py","w",  encoding="utf-8") as latest:
    latest.write(version)

setuptools.setup(
    name="threefive",
    version=version,
    author="Adrian and a cast of thousands.",
    author_email="spam@iodisco.com",
    description="Pythonic SCTE35",
    long_description=readme,
    long_description_content_type="text/markdown",
    url="https://github.com/futzu/threefive",
    install_requires=[
        'new_reader >= 0.1.3',
        "pyaes",
    ],

    scripts=['bin/threefive'],
    packages=setuptools.find_packages(),
    classifiers=[
        "License :: OSI Approved :: MIT License",
        "Operating System :: OS Independent",
        "Programming Language :: Python :: 3.6",
        "Programming Language :: Python :: Implementation :: PyPy",
        "Programming Language :: Python :: Implementation :: CPython",
    ],
    python_requires=">=3.6",
)
```

**Comment by futzu** [comment]
That doesn't work as I want yet.

**Comment by futzu** [comment]
new idea.

version.py just has version='2.3.79'
and we read it like a regular file in setup.py
then instead of calling threefive.version(), 
```py3
>>> from threefive.version import version
>>> version
'2.3.79'
```

```py3
#!/usr/bin/env python3

import setuptools

with open("README.md", "r", encoding="utf-8") as fh:
    readme = fh.read()

with open("threefive/version.py","r", encoding="utf-8") as latest:
    version = latest.read().split("'")[1]

setuptools.setup(
    name="threefive",
    version=version,
    author="Adrian and a cast of thousands.",
    author_email="spam@iodisco.com",
    description="Pythonic SCTE35",
    long_description=readme,
    long_description_content_type="text/markdown",
    url="https://github.com/futzu/threefive",
    install_requires=[
        'new_reader >= 0.1.7',
        "pyaes",
    ],

    scripts=['bin/threefive'],
    packages=setuptools.find_packages(),
    classifiers=[
        "License :: OSI Approved :: MIT License",
        "Operating System :: OS Independent",
        "Programming Language :: Python :: 3.6",
        "Programming Language :: Python :: Implementation :: PyPy",
        "Programming Language :: Python :: Implementation :: CPython",
    ],
    python_requires=">=3.6",
)
```


Does that work for you?

**Comment by futzu** [comment]
```py3
a@debian:~$ pypy3
Python 3.9.16 (7.3.11+dfsg-2, Feb 06 2023, 16:52:03)
[PyPy 7.3.11 with GCC 12.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>>> import threefive
>>>> threefive.version
'2.3.79'
```

**Comment by airlangga-m2amedia** [comment]
Yep that works for me. There is also a way of including the version file with setuptools, potentially by using a MANIFEST.in file, but this way works

**Comment by futzu** [comment]
Please don't take this the wrong way, I am not trying to step on your toes.

I should have tested that before making the package. I made the mistake and had to fix it quickly.

 This is your idea, I just changed the version  file location. 
 You did a good job, and you caught a bug that I have missed for a couple of years.
 **I encourage you to submit a patch anytime.**
  

I liked the VERSION file to be honest with you,
I don't really want to start using a manifest file.

The reason I need this to work is so people can easily report the version to me
when they ask for help.  I do a new release anytime a bug is fixed or a feature is added, 
so the version number is sometimes really important.

What is super weird, 
when I did
```py3
with open("threefive/latest.py","w",  encoding="utf-8") as latest:
    latest.write(version)
```
python  had a problem with that because it was just `2.3.79` in lastest.py
so then I tried to write an fstring with the version quoted like 

```py3
latest.write(f' "{version}" ')
```
and threw an error because the default toml format changed the version to 
```
"2.3.79
"
```
I have no idea why. I escaped the quotes, I did it without an f string, same thing every time.

I don't spend a lot of time on packaging, if you have ideas, let's do them. 

I also really want to do the newer style functions. 

I want to hear more of what you have to say.

Adrian

**Comment by futzu** [comment]
One other thing you got me thinking about, we need a writer like new_reader. 
I don't if you've looked at it but it handles files, http(s), UDP unicast and multicast all the same way.
If we had a writer that just handled the encoding and opening so we could do 
```py3
with writer("some_file") as wtr:
    wtr.write(stuff)

```

**Comment by airlangga-m2amedia** [comment]
Absolutely no worries. Apologies I didn't test against the use case you quoted above. I'm all for the solution you have merged in.

**Comment by airlangga-m2amedia** [comment]
Initially as I was doing this fix, I was looking at implementing PEP517 and PEP518 for the threefive repo. Did you have any thoughts or preferences on this?

**Comment by futzu** [comment]
I've been wanting to look into new packaging, I never heard of pep517 and pep518 before, I just scanned over them and they look fine to me. If you think it's better, let's try it. The only thing is I may need you to help get my other repos using it as well. 

I recently had a few disagreements with the Pep folks over pep1 and pep 668, those guys do not take criticism well. :)

**Comment by airlangga-m2amedia** [comment]
I'm a bit busy this week, but I'll try and pull together a PR for you next week (10th April onwards) and see what you make of it.

**Pull request merged**